microsoft/PowerToys
# Turn on C++ code analysis

Using Cpp.Build.props to setup C++ code analysis as default. Starting with a ruleset based on Microsoft Recomended Native rules. But changing all to **info**. Future pull request should change a rule from info to **error** and fix the code.


## Summary of the Pull Request


## PR Checklist

- [X] **Starts the work to get to:** https://github.com/microsoft/PowerToys/issues/940
- [X] **Tests:** Added/updated and all pass. No new test or update was done.
- [X] **Localization:** All end user facing strings can be localized. No changes to any string


## Detailed Description of the Pull Request / Additional comments
This PR turns on Code Analysis for C++ code through Cpp.Build.props files.
It also add a CppRuleSet.ruleset that is based on Microsoft Native Recommended Rules, but with all Rules set to **info** (no **warning** and no **error**).



## Validation Steps Performed

Another 2 pull request were created:
- one to test that setting a rule to **error** would make the CI break. #21183
- another to fix the errors and make sure the CI passes. #21163
-
